modules/posts/includes/class-post-model-base.php

```diff
@@ -267,8 +408,550 @@ function restore_postdata( $postdata ) {
+			 *
+			 * 'name' is too generic so 'post_name' is accessed via slug():
+			 *
+			 *       $this->slug()              => return $this->_post->post_name
+			 *
+			 * 'post_type' is too iconic so post_type() is used vs. type():
+			 *
+			 * 'post_parent' is accessed by parent_id():
+			 *
+			 *       $this->parent_id()         => return $this->_post->post_parent
+			 *
+			 * Others are accessed as method name sans 'post_' prefix, i.e.
+			 *
+			 *       $this->ID()                => return $this->_post->ID
+			 *       $this->menu_order()        => return $this->_post->menu_order
+			 *       $this->date()              => return $this->_post->post_date
+			 *       $this->date_gmt()          => return $this->_post->post_date_gmt
+			 *       $this->modified()          => return $this->_post->post_modified
+			 *       $this->modified_gmt()      => return $this->_post->post_modified_gmt
+			 *       $this->password()          => return $this->_post->password
+			 *       $this->comment_count()     => return $this->_post->comment_count
+			 *       $this->pinged()            => return $this->_post->pinged
+			 *       $this->to_ping()           => return $this->_post->to_ping
+			 *       $this->ping_status()       => return $this->_post->ping_status
+			 *       $this->comment_status()    => return $this->_post->comment_status
+			 *
+			 * Lastly 'post_type', 'post_content' and 'post_excerpt' are accessed via more functions, so:
+			 *
+			 *       $this->post_type()         !=> return $this->_post->post_type
+			 *       $this->content()           !=> return $this->_post->post_content
+			 *       $this->excerpt()           !=> return $this->_post->post_excerpt
+			 *
+			 */
+
+			switch ( preg_replace( '#^post_(.+)$#', '$1', $method_name ) ) {
+
+				case 'ID':
+				case 'menu_order';
+				case 'comment_count':
+
+					$data_type     = 'int';
+					$property_name = $method_name;
+					break;
+
+				case 'date':
+				case 'date_gmt':
+				case 'modified':
+				case 'modified_gmt':
+
+					$data_type     = 'date';
+					$property_name = "post_{$method_name}";
+					break;
+
+				case 'status':
+				case 'password':
+
+					$data_type     = 'string';
+					$property_name = "post_{$method_name}";
+					break;
+
+				case 'comment_status':
+				case 'ping_status':
+				case 'to_ping':
+				case 'pinged':
+					$data_type     = 'string';
+					$property_name = $method_name;
+					break;
+
+			}
+
+			if ( ! $property_name ) {
+
+				$value = parent::__call( $method_name, $args );
+
+			} else {
+
+				$value = $this->_post->$property_name;
+
+				switch ( $data_type ) {
+					case 'int':
+
+						$value = intval( $value );
+						break;
+
+					case 'date':
+						/**
+						 * @todo Verify that this is what we want to standardize on.
+						 */
+						$value = mysql2date( DATE_W3C, $value );
+
+					default:
+						/*
+						 * No need to do anything for a string.
```

Comment:
Are there any other data types that might hit this default statement?  Like arrays, objects, floats, etc.?